spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/PropertiesConverter.java

```diff
@@ -5,11 +5,74 @@
 
 package com.datastrato.gravitino.spark.connector;
 
+import java.util.HashMap;
 import java.util.Map;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
-/** Transform table properties between Gravitino and Spark. */
+/** Interface for transforming properties between Gravitino and Spark. */
 public interface PropertiesConverter {
+
+  public static final String SPARK_PROPERTY_PREFIX = "spark.bypass.";
+
+  /**
+   * Converts properties from application provided properties and Gravitino catalog properties to
+   * Spark connector properties.
+   *
+   * <p>It provides the common implementation, include extract properties with "spark.bypass"
+   * prefix, merge user provided options and transformed properties.
+   *
+   * @param options Case-insensitive properties map provided by application configuration.
+   * @param properties Gravitino catalog properties.
+   * @return properties for the Spark connector.
+   */
+  default Map<String, String> toSparkCatalogProperties(
+      CaseInsensitiveStringMap options, Map<String, String> properties) {
+    Map<String, String> all = new HashMap<>();
+    if (properties != null) {
+      properties.forEach(
+          (k, v) -> {
+            if (k.startsWith(SPARK_PROPERTY_PREFIX)) {
+              String newKey = k.substring(SPARK_PROPERTY_PREFIX.length());
+              all.put(newKey, v);
+            }
+          });
+    }
+
+    Map<String, String> transformedProperties = transformSparkCatalogProperties(properties);
+    if (transformedProperties != null) {
+      all.putAll(transformedProperties);
+    }
+
+    if (options != null) {
+      all.putAll(options);
+    }
+    return all;
+  }
+
+  /**
+   * Transform properties from Gravitino catalog properties to Spark connector properties.
+   *
+   * <p>This interface focus on the catalog specific transform logic, the common logic are
+   * implemented in {@code toSparkCatalogProperties}.
+   *
+   * @param properties Gravitino catalog properties.
+   * @return properties for the Spark connector.
+   */
+  Map<String, String> transformSparkCatalogProperties(Map<String, String> properties);
```

Comment:
What is the meaning of `transform`, why don't you use `toXXX` to align with below interfaces if the semantics are the same.

Reply:
done